src/main/java/com/shilymily/bms/service/CompanyService.java

```diff
@@ -0,0 +1,37 @@
+package com.shilymily.bms.service;
+
+import com.shilymily.bms.entity.Company;
+import com.shilymily.bms.repository.CompanyRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by hmaskai on 9/8/18.
+ */
+@Service
+public class CompanyService {
+
+    @Autowired
+    private CompanyRepository companyRepository;
+
+    public void addOrUpdateCompany(Company company){
+        companyRepository.save(company);
```

Comment:
Would suggest some basic validations here, I see null check is already taken care off, but you could add validations for correct phone number or email-id and pan-card.
What's the use of storing data if it is wrong and doesn't provide any information.

Reply:
Its a good point but these checks need to be done in the front-end microservice

Reply:
Agreed, as long as it is taken care of somewhere.